### PR TITLE
build(mage): ➕ add mage task for exporting plugin docs to markdown

### PR DESCRIPTION
**Pull request by sheldonhull**
- Add mage task for exporting the plugin usage information to docs directory.
- Add reference to this on the readme.

In future PR, let's add a full example like mentioned in #45. 
This makes it easier to find the ansible plugin docs without CLI, but won't complete it till we have the full example included.

### CODE REVIEW

File: magefile.go [hunk 1]
```diff
@@ -565,3 +573,25 @@ func checkEnvVar(ckv *checkEnv) (string, pterm.TableData, error) {
...
+		if err := venvInstall("ansible-doc-extractor"); err != nil {
+			return fmt.Errorf("failed to install ansible-doc-extractor: %w", err)
+		}
+	} else {
+		pterm.Info.Println("ansible-doc-extractor already installed")
+	}
+	if err := mkdir(DocDirectory); err != nil {
+		return fmt.Errorf("failed to create doc directory: %w", err)
+	}
+	return venvRunBinary(true,
```
Comment (andrii-zakurenyi) [inline magefile.go:591]:
```suggestion
	return venvRunV(
```

File: magefile.go [hunk 2]
```diff
@@ -565,3 +573,25 @@ func checkEnvVar(ckv *checkEnv) (string, pterm.TableData, error) {
 		return "", nil, nil // Unreachable.
 	}
 }
+
+// ExportPluginDocs runs ansible-doc-extractor to generate a markdown file of the plugin documentation.
+func ExportPluginDocs() error {
```
Comment (andrii-zakurenyi) [inline magefile.go:578]: I've tried to organize this file using following structure:
1. Imports
2. Constants
3. Mage targets
4. Helper functions called by mage targets

There is even a header which denotes the beginning of helper functions block.
```go
// ----------------------------------- //
//          Helper Functions           //
// ----------------------------------- //
```

Please either remove that header and we will agree to not use that structure or place this new function above that header :)

**Review by andrii-zakurenyi — APPROVED** [review]
Minor comments. Other than that LGTM

File: magefile.go (hunk 2 above)
Comment (sheldonhull) [reply magefile.go:578]: That's fine by me. I have sidebar that auto sorts so I never noticed. I'll push that later and merge. Cheers.

Comment (sheldonhull) [reply magefile.go]: Finally fixed. I thought about it before and would love a linter that did this. Not sure I want to try and keep all my files sorted  it makes navigation manually nicer for sure. 

Be a cool linting project.

**Pull request merged**